aten/src/ATen/core/ATenDispatch.h

```diff
@@ -15,20 +18,34 @@
 
 namespace at {
 
+template <typename R, typename... Args>
+struct lazy;
+template <typename R, typename... Args>
```

Comment:
I can't follow what is going on here. Can you document it? Why is this working with raw char* schema strings?